node/src/main/scala/coop/rchain/node/node.scala

```diff
@@ -250,31 +250,31 @@ class NodeRuntime(conf: Configuration, host: String)(implicit scheduler: Schedul
         case handled       => handled.pure[Effect]
     }
 
-    for {
+    val info: Effect[Unit] = for {
       _ <- Log[Effect].info(
             s"RChain Node ${BuildInfo.version} (${BuildInfo.gitHeadCommit.getOrElse("commit # unknown")})")
+      _ <- if (conf.server.standalone) Log[Effect].info(s"Starting stand-alone node.")
+          else Log[Effect].info(s"Starting node that will bootstrap from ${conf.server.bootstrap}")
+    } yield ()
+
+    val loop: Effect[Unit] = for {
+      _ <- Connect.clearConnections[Effect]
+      _ <- Connect.findAndConnect[Effect](Connect.connect[Effect] _)
+      _ <- requestApprovedBlock[Effect]
+      _ <- time.sleep(5000).toEffect
+    } yield ()
+
+    for {
+      _       <- info
       servers <- acquireServers(runtime)
       _       <- addShutdownHook(servers, runtime, casperRuntime).toEffect
       _       <- startServers(servers)
       _       <- startReportJvmMetrics.toEffect
       _       <- TransportLayer[Effect].receive(handleCommunication)
+      ndFiber <- NodeDiscovery[Task].discover.forever.fork.toEffect
       _       <- Log[Effect].info(s"Listening for traffic on $address.")
-      res <- ApplicativeError_[Effect, CommError].attempt(
-              if (conf.server.standalone) Log[Effect].info(s"Starting stand-alone node.")
-              else
-                Connect.connectToBootstrap[Effect](conf.server.bootstrap,
-                                                   maxNumOfAttempts = 5,
-                                                   defaultTimeout = defaultTimeout))
-      _ <- if (res.isRight)
-            MonadOps.forever((i: Int) =>
-                               Connect
-                                 .findAndConnect[Effect](defaultTimeout)
-                                 .apply(i) <* requestApprovedBlock[Effect],
-                             0)
-          else ().pure[Effect]
-      _ <- exit0.toEffect
+      _       <- loop.forever
```

Comment:
We have now 2 forever loops with a sleep delay. Maybe it's time to introduce a `forever` with delay parameter?